PyTorch_dcgan_lsun/README.md

```diff
@@ -0,0 +1,116 @@
+# PyTorch - DCGAN
+
+This example is based on [this implementation](https://github.com/pytorch/examples/tree/master/dcgan)
+
+
+## Downloading the dataset
+
+You can download the LSUN dataset by cloning [this repo](https://github.com/fyu/lsun) and running
+```
+python download.py -c bedroom
+```
+
+## Making the code parallel
+
+It is actually very easy to allow PyTorch code to run on multiple GPUs
+
+### Locally
+
+All you need to do to make a data parallel model that can be run on multiple GPUs locally:
+```
+from torch import nn
+<your code including creating the model>
+model = nn.DataParallel(model)
+```
+That's it. After that you just control which GPUs you would like your code to run on by specifing their number in CUDA_VISIBLE_DEVICES before your command (tensorhive takes care of that for you when you specify the devices while creating the task).
+
+### Distributed
+
+There is not that much more to do in this case.
+
+```
+from torch import distributed, nn
+
+
+<argument parser>
+
+parser.add_argument('--backend', type=str, default='gloo', help='Name of the backend to use.')
+parser.add_argument(
+    '-i',
+    '--init-method',
+    type=str,
+    default='tcp://127.0.0.1:23456',
+    help='URL specifying how to initialize the package.')
+parser.add_argument('-s', '--world-size', type=int, default=1, help='Number of processes participating in the job.')
+parser.add_argument('-r', '--rank', type=int, default=0, help='Rank of the current process.')
+
+
+if opt.world_size > 1:
+        distributed.init_process_group(
+            backend=opt.backend,
+            init_method=opt.init_method,
+            world_size=opt.world_size,
+            rank=opt.rank,
+        )
+
+
+<code with model creation, in this case Generator is called netG>
+netG = nn.parallel.DistributedDataParallel(netG)
+
+<another model creation in this case discriminator called netD>
+netD = nn.parallel.DistributedDataParallel(netD)
+```
```

Comment:
Instead of indications how to change the code, a patch should be provided. See https://github.com/roscisz/dnn_training_benchmarks/blob/master/TensorFlowV1_DeepSpeech_ldc93s1/deepspeech_benchmarking.patch
This way, changes can be introduced easily, and if someone wants to examine and understand them, they will.

Reply:
But wasn't this a problem with the DeepSpeech example that it tied us to specific version of the used example?

Reply:
We want our example to be reproducible regardless of changes in the utilized libraries, so we should specify versions for which we checked that it works, and update them as often as possible. This includes version of the training code, so we should provide a specific commit in the DCGAN repo.

The problem with DeepSpeech was that it required a specific version of a binary library hosted by Mozilla, which is no longer hosted. We are temporarily hosting the obsolete version, but updating that example to the latest version seems to be a better solution (BTW I will post an issue regarding that).

Reply:
Added the patch and changed the instructions accordingly.